components/common/src/main/java/com/hotels/styx/config/schema/Schema.java

```diff
@@ -217,12 +325,42 @@ public FieldValue elementType() {
         public FieldValue elementType() {
             return elementType;
         }
+
+        @Override
+        public Optional<String> isCorrectType(List<String> parents, JsonNode parent, JsonNode element, Function<String, FieldValue> schemas) {
+            return element.isObject() ? validateMap(parents, parent, element, schemas) : message(parents, element);
+        }
+
+        @Override
+        public String describe() {
+            return format("MAP (%s)", elementType.describe());
+        }
+
+        private Optional<String> validateMap(List<String> parents, JsonNode parent, JsonNode mapNode, Function<String, FieldValue> schemas) {
+            AtomicReference<Optional<String>> success = new AtomicReference<>(Optional.empty());
+
+            mapNode.fieldNames().forEachRemaining(key -> {
+                JsonNode entry = mapNode.get(key);
+                Optional<String> correctType = elementType.isCorrectType(push(parents, format("%s", key)), mapNode, entry, schemas);
+
+                if (correctType.isPresent()) {
+                    success.set(correctType);
```

Comment:
This doesn't seem right. Won't a correctly typed entry override a previous incorrectly typed entry to make the map "correct" overall?

Reply:
Renamed local variables. Hopefully the code logic is clearer now.